finagle-redis/src/main/scala/com/twitter/finagle/redis/protocol/commands/Sentinels.scala

```diff
@@ -0,0 +1,169 @@
+
+object SentinelReset extends SentinelHelper {
+  val command = "RESET"
+  def apply(args: Seq[String]): SentinelReset = {
+    val name = trimList(args, 2, "SENTINEL RESET")(0)
+    new SentinelReset(name)
+  }
+}
+
+case class SentinelFailover(name: String)
+  extends Sentinel(SentinelFailover.command, Seq(name))
+
+object SentinelFailover extends SentinelHelper {
+  val command = "FAILOVER"
+  def apply(args: Seq[String]): SentinelFailover = {
+    val name = trimList(args, 2, "SENTINEL FAILOVER")(0)
+    new SentinelFailover(name)
+  }
+}
+
+case class SentinelCkQuorum(name: String)
+  extends Sentinel(SentinelCkQuorum.command, Seq(name))
+
+object SentinelCkQuorum extends SentinelHelper {
+  val command = "CKQUORUM"
+  def apply(args: Seq[String]): SentinelCkQuorum = {
+    val name = trimList(args, 2, "SENTINEL CKQUORUM")(0)
+    new SentinelCkQuorum(name)
+  }
+}
+
+case class SentinelFlushConfig()
+  extends Sentinel(SentinelFlushConfig.command, Nil)
+
+object SentinelFlushConfig extends SentinelHelper {
+  val command = "FLUSHCONFIG"
+  def apply(args: Seq[String]): SentinelFlushConfig = {
+    new SentinelFlushConfig()
+  }
+}
+
+case class SentinelMonitor(name: String, ip: String, port: Int, quorum: Int)
+  extends Sentinel(SentinelMonitor.command, Seq(name, ip, port.toString, quorum.toString))
+
+object SentinelMonitor extends SentinelHelper {
+  val command = "MONITOR"
+  def apply(args: Seq[String]): SentinelMonitor = {
+    trimList(args, 2, "SENTINEL MONITOR") match {
+      case name :: ip :: port :: quorum :: Nil =>
+        new SentinelMonitor(name, ip, port.toInt, quorum.toInt)
+    }
+  }
+}
+
+case class SentinelRemove(name: String)
+  extends Sentinel(SentinelRemove.command, Seq(name))
+
+object SentinelRemove extends SentinelHelper {
+  val command = "REMOVE"
+  def apply(args: Seq[String]): SentinelRemove = {
+    val list = trimList(args, 2, "SENTINEL REMOVE")
+    new SentinelRemove(list(0))
+  }
+}
+
+case class SentinelSet(name: String, option: String, value: String)
+  extends Sentinel(SentinelSet.command, Seq(name, option, value))
+
+object SentinelSet extends SentinelHelper {
+  val command = "SET"
+  def apply(args: Seq[String]): SentinelSet = {
+    trimList(args, 2, "SENTINEL SET") match {
+      case name :: option :: value :: Nil =>
+        new SentinelSet(name, option, value)
+    }
+  }
+}
+
+trait SentinelHelper {
+  def command: String
+  def apply(args: Seq[String]): Sentinel
+}
+
+abstract class Sentinel(sub: String, args: Seq[String]) extends Command {
+  def command = Commands.SENTINEL
+  def toChannelBuffer = RedisCodec.toUnifiedFormat(
+      Seq(CommandBytes.SENTINEL, StringToChannelBuffer(sub)) ++
+        args.map(StringToChannelBuffer(_)))
+}
+
+object Sentinel {
+
+  val subCommands: Seq[SentinelHelper] = Seq(
+    SentinelMaster, SentinelMasters, SentinelMonitor, SentinelRemove, SentinelSet)
+
+  def fromBytes(args: Seq[Array[Byte]]): Sentinel = {
+    apply(args.map(new String(_)))
```

Comment:
This is going to use the platform's default charset when constructing the string. I'm not sure if this is an actual problem in this scenario or not, but it would likely be best to make it explicit (e.g. UTF-8).